### PR TITLE
RDM-9551-TF-DO-NOT-MERGE

### PR DESCRIPTION
**Pull request by nathan-clark**
Upgrade Terraform
from plan: 
# module.vault.azurerm_key_vault_access_policy.managed_identity_access_policy will be destroyed but recreated is this ok?
~ default_action             = "Allow" -> "Deny"
but i think it should be deny anyhow?!?

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

File: storage-account.tf [hunk 1]
```diff
@@ -3,18 +3,16 @@
...
-  resource_group_name       = "${azurerm_resource_group.rg.name}"
-  location                  = "${var.location}"
-  account_kind              = "StorageV2"
-  account_tier              = "Standard"
-  account_replication_type  = "LRS"
-  access_tier               = "Hot"
-
-  enable_blob_encryption    = true
-  enable_file_encryption    = true
+  source                   = "git@github.com:hmcts/cnp-module-storage-account.git?ref=azurermv2"
```
Comment (adusumillipraveen) [inline storage-account.tf:6]:
```suggestion
  source                   = "git@github.com:hmcts/cnp-module-storage-account?ref=azurermv2"
```

File: storage-account.tf [hunk 2]
```diff
@@ -68,18 +66,16 @@ output "storage_account_name" {
...
```
Comment (adusumillipraveen) [inline storage-account.tf:69]:
```suggestion
  source                   = "git@github.com:hmcts/cnp-module-storage-account?ref=azurermv2"
```

File: appinsights.tf [hunk 3]
```diff
@@ -1,8 +1,8 @@
 resource "azurerm_application_insights" "appinsights" {
   name                = "${var.product}-${var.env}"
-  location            = "${var.location}"
-  resource_group_name = "${azurerm_resource_group.rg.name}"
-  application_type    = "${var.application_type}"
```
Comment (adusumillipraveen) [inline appinsights.tf:5]: Missed life cycle change in here?

Comment (nathan-clark) [reply appinsights.tf:5]: yes indeed @adusumillipraveen -ta

**Pull request closed without merging**